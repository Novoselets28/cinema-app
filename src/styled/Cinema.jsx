import { Button } from "react-bootstrap";
import styled from "styled-components";

export const CinemaScreen = styled.div`
  background-color: #000;
  color: #fff;
  text-align: center;
  font-size: 24px;
  padding: 10px;
  margin-bottom: 20px; /* Adjust margin as needed */
  /* You can also add additional styles to customize the appearance of the screen */
`;

export const StyledButton = styled(Button)`
  background-color: green;
  color: white;
`;