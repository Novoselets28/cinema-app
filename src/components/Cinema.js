import React, { useEffect, useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';

import '../index.css';
import { Row, Col, Button } from 'react-bootstrap';
import {CinemaScreen, HomeButton, StyledButton, StyledContainer} from '../styled/Cinema';
import { useDispatch, useSelector } from 'react-redux';

const Cinema = () => {
  const { date, selectedSession } = useParams();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const availableSeats = useSelector((state) => state.cinema.availableSeats);
  const selectedSeats = useSelector((state) => state.cinema.selectedSeats);


  const [isAlertActive, setIsAlertActive] = useState(false);


  useEffect(() => {
    // Dispatch an action to fetch available seats.
    dispatch({ type: 'FETCH_AVAILABLE_SEATS' });
  }, [dispatch]);


  const isSeatBooked = (seat) => {
    return selectedSeats.includes(seat);
  };

  const handleSeatSelect = (seat) => {
      dispatch({ type: 'TOGGLE_SELECTED_SEAT', payload: seat });

  };

  const handleBookSeats = () => {
    alert(`Booked seats: ${selectedSeats.join(', ')} for ${selectedSession}`);
    setIsAlertActive(true)
  };

  const areSeatsSelected = selectedSeats.length > 0;

  const handleGoMainPage = () => {
    navigate('/');
  };


  return (
    <StyledContainer>
      <h4>You selected film for {date}</h4>
      <h2>Just choose a seat</h2>
      <Row className="seat-container">
      <CinemaScreen className="cinema-screen">Cinema Screen</CinemaScreen>
        {availableSeats.map((seat) => (
          <Col key={seat} xs={2}>
            <StyledButton
              variant={isSeatBooked(seat) ? 'success' : 'secondary'}
              onClick={() => handleSeatSelect(seat)}
              className="mb-2"
            >
              {seat}
            </StyledButton>
          </Col>
        ))}
      </Row>
      <div className="confirm-btn">
        <Button 
        variant="primary" 
        onClick={handleBookSeats}
        disabled={!areSeatsSelected}
        >
        {areSeatsSelected ? 'Toggle Selected Seats Success' : 'No Seats Selected'}
        </Button>
        {isAlertActive && (
          <HomeButton variant="info" onClick={handleGoMainPage}>
            Home
          </HomeButton>
                  )}
      </div>
    </StyledContainer>
  );
};

export default Cinema;

