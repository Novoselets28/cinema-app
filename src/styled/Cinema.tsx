// import { Button } from 'react-bootstrap';
// import styled from 'styled-components';
// import { colors, fontSize } from './global';

// export const CinemaScreen = styled.div`
//   background-color: ${colors.secondary};
//   text-align: center;
//   font-size: ${fontSize.medium};
//   padding: 10px;
//   margin-bottom: 20px;
// `;

// export const HomeButton = styled(Button)`
//   margin: 5px;
// `;

// export const StyledContainer = styled.div`
//   margin-top: 5px;
// `

// Import necessary dependencies


import { Button } from 'react-bootstrap';
import styled from 'styled-components';
import { colors, fontSize } from './global';

// Define TypeScript types for props if needed
// For example: interface CinemaScreenProps {}

// Define your styled components
export const CinemaScreen = styled.div`
  background-color: ${colors.secondary};
  text-align: center;
  font-size: ${fontSize.medium};
  padding: 10px;
  margin-bottom: 20px;
`;

export const HomeButton = styled(Button)`
  margin: 5px;
`;

export const StyledContainer = styled.div`
  margin-top: 5px;
`;

export const Container = styled.div`
  // Your styles here
`;


