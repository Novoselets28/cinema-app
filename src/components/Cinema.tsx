import React, { useEffect, useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Row, Col, Button, Container } from 'react-bootstrap';

import { CinemaScreen, HomeButton, StyledContainer } from '../styled/Cinema';
import { fetchAvailableSeats } from '../redux/ducks/cinema';

interface CinemaProps {
  date: string;
  selectedSession: string;
}

const Cinema: React.FC<CinemaProps> = () => {
  const { date, selectedSession } = useParams();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const availableSeats:string[] = useSelector((state: any) => state.cinema.availableSeats) || [];
  const selectedSeats:string[] = useSelector((state: any) => state.cinema.selectedSeats);

  const [isAlertActive, setIsAlertActive] = useState<boolean>(false);

  useEffect(() => {
    dispatch(fetchAvailableSeats([]));
  }, [dispatch]);

  const isSeatBooked = (seat: string):boolean => selectedSeats.includes(seat);

  const handleSeatSelect = (seat:string):void => {
    dispatch({ type: 'TOGGLE_SELECTED_SEAT', payload: seat });
  };

  const handleBookSeats = ():void => {
    alert(`Booked seats: ${selectedSeats.join(', ')} for ${selectedSession}`);
    setIsAlertActive(true);
  };

  const areSeatsSelected:boolean = selectedSeats?.length > 0;

  const handleGoMainPage = (): void => {
    navigate('/');
  };

  return (
    <Container>
      <h4>You selected a film for {date}</h4>
      <h2>Just choose a seat</h2>
      <Row>
        <CinemaScreen>Cinema Screen</CinemaScreen>
        {availableSeats.map((seat) => (
          <Col key={seat} xs={2}>
            <Button
              variant={isSeatBooked(seat) ? 'success' : 'secondary'}
              onClick={() => handleSeatSelect(seat)}
              className="mb-2"
            >
              {seat}
            </Button>
          </Col>
        ))}
      </Row>
      <StyledContainer>
        <Button
          variant="primary"
          onClick={handleBookSeats}
          disabled={!areSeatsSelected}
        >
          {areSeatsSelected ? 'Toggle Selected Seats Success' : 'No Seats Selected'}
        </Button>
        {isAlertActive && (
          <HomeButton variant="info" onClick={handleGoMainPage}>
            Home
          </HomeButton>
        )}
      </StyledContainer>
    </Container>
  );
};

export default Cinema;