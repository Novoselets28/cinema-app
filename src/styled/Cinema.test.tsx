import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { CinemaScreen, HomeButton, StyledContainer } from './Cinema';
import { colors, fontSize } from './global';

describe('Styled Components', () => {
  it('CinemaScreen should have expected styles', () => {
    render(<CinemaScreen data-testid="cinema-screen" />);
    const cinemaScreen = screen.getByTestId('cinema-screen');

    // Use Testing Library's querying methods
    expect(cinemaScreen).toHaveStyle(`
      background-color: ${colors.secondary};
      text-align: center;
      font-size: ${fontSize.medium};
      padding: 10px;
      margin-bottom: 20px;
    `);

    // Add snapshot assertion
    expect(cinemaScreen).toMatchSnapshot();
  });

  it('HomeButton should have expected styles', () => {
    render(<HomeButton data-testid="home-button" />);
    const homeButton = screen.getByTestId('home-button');

    // Use Testing Library's querying methods
    expect(homeButton).toHaveStyle(`
      margin: 5px;
    `);

    // Add snapshot assertion
    expect(homeButton).toMatchSnapshot();
  });

  it('StyledContainer should have expected styles', () => {
    render(<StyledContainer data-testid="styled-container" />);
    const styledContainer = screen.getByTestId('styled-container');

    // Use Testing Library's querying methods
    expect(styledContainer).toHaveStyle(`
      margin-top: 5px;
    `);

    // Add snapshot assertion
    expect(styledContainer).toMatchSnapshot();
  });
});
