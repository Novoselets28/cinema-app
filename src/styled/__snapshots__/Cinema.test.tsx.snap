// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Components CinemaScreen should have expected styles 1`] = `
<div
  class="sc-aXZVg wnUcX"
  data-testid="cinema-screen"
/>
`;

exports[`Styled Components HomeButton should have expected styles 1`] = `
<button
  class="sc-gEvEer hnZgjw btn btn-primary"
  data-testid="home-button"
  type="button"
/>
`;

exports[`Styled Components StyledContainer should have expected styles 1`] = `
<div
  class="sc-eqUAAy fyYumI"
  data-testid="styled-container"
/>
`;
